contracts/contracts/RequestManager.sol

```diff
@@ -276,33 +276,33 @@ contract RequestManager is Ownable, LpWhitelist, RestrictedCalls {
 
         uint256 lpFeeTokenAmount = lpFee(amount);
         uint256 protocolFeeTokenAmount = protocolFee(amount);
-        uint256 totalTokenAmount = amount +
```

Comment:
Out of curiosity, what was the reason for removing the `totalTokenAmount` here? Is that also a part of the optimizations?

Reply:
I think the idea is that we don't use a storage slot, and instead just calculate it directly based on values that we already have.

Reply:
Aha, now it makes sense to me. Thanks!

Reply:
A function only has 16 slots on the stack (for parameters and local variables). In this function we reached the maximum amount but read storage variables multiple times. So in order to copy storage values into memory and save costs for reading from storage I had to free up some variables, like `totalTokenAmount`

Reply:
Great explainer! thanks.